src/main/java/appeng/util/item/AESharedItemStack.java

```diff
@@ -0,0 +1,210 @@
+/*
+ * This file is part of Applied Energistics 2.
+ * Copyright (c) 2013 - 2014, AlgorithmX2, All rights reserved.
+ *
+ * Applied Energistics 2 is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Applied Energistics 2 is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Applied Energistics 2.  If not, see <http://www.gnu.org/licenses/lgpl>.
+ */
+
+package appeng.util.item;
+
+
+import java.util.Objects;
+
+import com.google.common.base.Preconditions;
+
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemStack;
+import net.minecraft.nbt.NBTTagCompound;
+
+import appeng.api.config.FuzzyMode;
+import appeng.api.storage.data.IAEStackSearchKey;
+
+
+final class AESharedItemStack implements IAEStackSearchKey<ItemStack>
+{
+	private static final NBTTagCompound LOW_TAG = new NBTTagCompound();
+	private static final NBTTagCompound HIGH_TAG = new NBTTagCompound();
+
+	private final ItemStack itemStack;
+	private final int itemId;
+
+	public AESharedItemStack( final ItemStack itemStack )
+	{
+		this.itemStack = itemStack;
+		this.itemId = Item.getIdFromItem( itemStack.getItem() );
+	}
+
+	@Override
+	public ItemStack getDefinition()
+	{
+		return this.itemStack;
+	}
+
+	@Override
+	public int hashCode()
+	{
+		return Objects.hash( itemStack.getItem(), itemStack.getItemDamage() );
+	}
+
+	@Override
+	public boolean equals( final Object obj )
+	{
+		if( obj instanceof AESharedItemStack )
+		{
+			final AESharedItemStack other = (AESharedItemStack) obj;
+
+			Preconditions.checkState( this.itemStack.getCount() == 1, "ItemStack#getCount() has to be 1" );
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "ItemStack#getCount() has to be 1" 6 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1192)